api/v1/submissions/PKPSubmissionController.php

```diff
@@ -1641,6 +1658,49 @@ public function addDecision(Request $illuminateRequest): JsonResponse
         return response()->json(Repo::decision()->getSchemaMap()->map($decision), Response::HTTP_OK);
     }
 
+    /**
+     * Checks if sensitive review assignment data should be anonymized for authors and reviewers
+     *
+     * @param LazyCollection<Submission>|Submission $submissions the list of submissions with IDs as keys or a single submission
+     * @param ?LazyCollection<ReviewAssignment> $reviewAssignments
+     *
+     * @return false|Collection List of review IDs to anonymize or false;
+     */
+    public function anonymizeReviews(LazyCollection|Submission $submissions, ?LazyCollection $reviewAssignments = null): false|Collection
+    {
+        $currentUser = $this->getRequest()->getUser();
+        $submissionIds = is_a($submissions, Submission::class) ? [$submissions->getId()] : $submissions->keys()->toArray();
+        $reviewAssignments = $reviewAssignments ?? Repo::reviewAssignment()->getCollector()->filterBySubmissionIds($submissionIds)->getMany();
+
+        $currentUserReviewAssignment = Repo::reviewAssignment()->getCollector()
+            ->filterBySubmissionIds($submissionIds)
+            ->filterByReviewerIds([$currentUser->getId()])
+            ->getMany()
+            ->first();
+
+        $currentUserStageAssignments = StageAssignment::withSubmissionIds($submissionIds)
+            ->withUserId($currentUser->getId())
+            ->get();
+
+        $isAuthor = $currentUserStageAssignments->contains(
+            function (StageAssignment $stageAssignment) {
+                $userGroup = Repo::userGroup()->get($stageAssignment->userGroupId);
+                return $userGroup->getRoleId() == Role::ROLE_ID_AUTHOR;
+            }
+        );
+
+        if ($currentUserReviewAssignment || $isAuthor) {
+            $anonymizeReviews = $reviewAssignments->map(function (ReviewAssignment $reviewAssignment, int $reviewId) use ($currentUserReviewAssignment) {
+                if ($currentUserReviewAssignment && $currentUserReviewAssignment->getId() == $reviewId) {
```

Comment:
Just to double check: It can be only one review assignment for one and the same user and submission, right?

Reply:
Thanks! Missed it - I've implemented the method to serve only single submission first. A reviewer can have more than one assignment in the list of submissions.